index.js

```diff
@@ -41,31 +41,34 @@ S3BlobStore.prototype.createReadStream = function (opts) {
   return stream;
 };
 
-S3BlobStore.prototype.uploadParams = function (opts) {
+S3BlobStore.prototype.baseParams = function (opts) {
   opts = Object.assign({}, opts, {
     params: Object.assign({}, opts.params)
   });
 
-  var filename = opts.name || opts.filename;
-  var key = opts.key || filename;
-  var contentType = opts.contentType;
+  const filename = opts.name || opts.filename;
+  const key = opts.key || filename;
+  let params = opts.params;
 
-  var params = opts.params;
   params.Bucket = params.Bucket || this.bucket;
   params.Key = params.Key || key;
 
-  if (!contentType) {
-    contentType = filename ? mime.lookup(filename) : mime.lookup(opts.key);
-  }
-  if (contentType) params.ContentType = contentType;
+  return params
+}
+
+S3BlobStore.prototype.uploadParams = function (opts) {
+  const params = this.baseParams(opts)
+  const filename = opts.name || opts.filename || params.Key;
+  let contentType = opts.contentType || mime.lookup(filename)
```

Comment:
I think `filename` is already present in `params.Key`,
as in `baseParams` function, you assign to `params.Key` either `opts.name`, `opts.filename` or `opts.key`.

```suggestion
  const contentType = opts.contentType || mime.lookup(params.Key)
```

Also, as `contentType` is defined once, I think `const` is better than `let` ?